crates/biome_js_analyze/src/lint/nursery/use_at_index.rs

```diff
@@ -0,0 +1,927 @@
+        {
+            if arg >= 0 {
+                return None;
+            }
+        }
+    }
+    Some(UseAtIndexState::new(
+        AnyJsExpression::JsUnaryExpression(member),
+        ErrorType::Index { is_negative: false },
+        object,
+    ))
+}
+
+/// check hoge[0]
+fn check_computed_member_expression(
+    exp: &JsComputedMemberExpression,
+    option: &UseAtIndexOptions,
+) -> Option<UseAtIndexState> {
+    // check slice
+    if let Some(slice_err) =
+        analyze_slice_element_access(&AnyJsExpression::JsComputedMemberExpression(exp.clone()))
+    {
+        return Some(slice_err);
+    }
+    // invalid optional chain, mutable case
+    if exp.is_optional_chain()
+        || is_within_delete_expression(&AnyJsExpression::JsComputedMemberExpression(exp.clone()))
+            .unwrap_or(false)
+    {
+        return None;
+    }
+    // check member
+    let member = solve_parenthesized_expression(exp.member().ok()?)?;
+    let object = exp.object().ok()?;
+    match member.clone() {
+        // hoge[hoge.length - 1]
+        AnyJsExpression::JsBinaryExpression(binary) => {
+            check_binary_expression_member(binary, object, option)
+        }
+        // hoge[1]
+        AnyJsExpression::AnyJsLiteralExpression(literal) => {
+            check_literal_expression_member(literal, object, option)
+        }
+        // hoge[-x]
+        AnyJsExpression::JsUnaryExpression(unary) => {
+            check_unary_expression_member(unary, object, option)
+        }
+        AnyJsExpression::JsIdentifierExpression(_) => None,
+        _ if option.check_all_index_access => Some(UseAtIndexState::new(
+            member,
+            ErrorType::Index { is_negative: false },
+            object,
+        )),
+        _ => None,
+    }
+}
+
+fn check_call_expression_last(
+    call_exp: &JsCallExpression,
+    member: &JsStaticMemberExpression,
+) -> Option<UseAtIndexState> {
+    let args: Vec<_> = call_exp
+        .arguments()
+        .ok()?
+        .args()
+        .into_iter()
+        .flatten()
+        .collect();
+    if args.len() != 1 {
+        return None;
+    }
+    let object = member.object().ok()?;
+    let AnyJsExpression::JsIdentifierExpression(object) = object else {
+        return None;
+    };
+    let lodash_function = ["_", "lodash", "underscore"];
+    let object_name = object.syntax().text().to_string();
+    if lodash_function.contains(&object_name.as_str()) {
+        let AnyJsCallArgument::AnyJsExpression(arg0) = args[0].clone() else {
+            return None;
+        };
+        Some(UseAtIndexState::new(
+            make_number_literal(-1),
+            ErrorType::GetLastFunction,
+            solve_parenthesized_expression(arg0)?,
+        ))
+    } else {
+        None
+    }
+}
+
+fn check_call_expression_char_at(
+    call_exp: &JsCallExpression,
+    member: &JsStaticMemberExpression,
+    option: &UseAtIndexOptions,
+) -> Option<UseAtIndexState> {
+    let args: Vec<_> = call_exp
```

Comment:
We could use `get_arguments_by_index`. See my other comment about `get_arguments_by_index`.